flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/TableEnvironment.scala

```diff
@@ -98,8 +98,42 @@ abstract class TableEnvironment(val config: TableConfig) {
 
     checkValidTableName(name)
 
-    val tableTable = new TableTable(table.getRelNode)
-    registerTableInternal(name, tableTable)
+    table.tableEnv match {
+      case e: BatchTableEnvironment =>
+        val tableTable = new TableTable(table.getRelNode)
+        registerTableInternal(name, tableTable)
+      case e: StreamTableEnvironment =>
+        val sTableTable = new TransStreamTable(table.getRelNode, true)
+        tables.add(name, sTableTable)
+    }
+
+  }
+
+  protected def registerStreamTableInternal(name: String, table: AbstractTable): Unit = {
+
+    if (isRegistered(name)) {
+      throw new TableException(s"Table \'$name\' already exists. " +
+        s"Please, choose a different name.")
+    } else {
+      tables.add(name, table)
+    }
+  }
+
+  /**
+   * Replaces a registered Table with another Table under the same name.
+   * We use this method to replace a [[org.apache.flink.api.table.plan.schema.DataStreamTable]]
+   * with a [[org.apache.calcite.schema.TranslatableTable]].
+   *
+   * @param name
+   * @param table
+   */
+  protected def replaceRegisteredStreamTable(name: String, table: AbstractTable): Unit = {
```

Comment:
rename to `replaceRegisteredTable()` since it is not specific for stream tables?